dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -198,9 +201,33 @@ private static void ProcessPositionalArguments(KernelFunctionMetadata functionMe
     private static object? ParseResult(FunctionResult result)
     {
         var resultAsObject = result.GetValue<object?>();
-        if (result.ValueType is not null && resultAsObject is not null && result.ValueType == typeof(OpenAIChatMessageContent))
+
+        // Extract content from wrapper types and deserialize accordingly
+        if (result.ValueType is not null && resultAsObject is not null)
         {
-            resultAsObject = ((OpenAIChatMessageContent)resultAsObject).Content;
+#pragma warning disable SKEXP0042 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+            if (resultAsObject is RestApiOperationResponse restApiOperationResponse)
```

Comment:
@dmytrostruk this is where I'm using the OpenApi package, do you have suggestions to do this without the package?

Reply:
@SergeyMenshykh is trying to move `RestApiOperationResponse` to Abstractions, so it can be re-used in multiple projects without adding `Functions.OpenApi` reference.

In this PR I did the same for `OpenAIChatMessageContent` and OpenAI project reference. Instead of `OpenAIChatMessageContent`, you can use `ChatMessageContent` with `Content` property:
https://github.com/microsoft/semantic-kernel/pull/4355

Reply:
Nvm I see your PR to resolve this #4355 [.Net: Removed Connectors.OpenAI dependency from PromptTemplates.Handlebars project ](https://github.com/microsoft/semantic-kernel/pull/4355)